src/swarmpal/toolboxes/secs/dsecs_algorithms.py

```diff
@@ -0,0 +1,1895 @@
+"""Algorithms for low latitude spherical elementary current system analysis.
+
+Adapted from MatLab code by Heikki Vanhamäki.
+
+"""
+import datetime as dt
+
+import numpy as np
+
+# from sub_fit_1D_DivFree import SwarmMag2J_test_fit_1D_DivFree
+import xarray as xr
+
+import swarmpal.toolboxes.secs.aux_tools as auto
+from swarmpal.toolboxes.secs import SecsInputs
+from swarmpal.toolboxes.secs.aux_tools import (
+    sph2sph,
+    sub_FindLongestNonZero,
+    sub_Swarm_grids,
+)
+
+
+def sub_fit_1D_DivFree(SwA, SwC, result):
+    # extract radii of the Earth and ionospheric current layer
+    Re = result["Re"]
```

Comment:
`F841 local variable 'Re' is assigned to but never used`

Would you prefer to leave these in? as we could argue it sometimes helps with readability. I could add a config to ignore such warnings.

Reply:
I think we can get rid of assigned by never used variables. I will try go go through them, I think some of them are related to the question what parameters we need to store in the dsecscdata to perform the analysis because it's done step by step. 

As a side note, I think this function is obsolete and should be removed, because the fittings have been moved to inside the dsecsdata class.